text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,895 @@
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetMessagesByAddressResponse"
+              examples:
+                default:
+                  $ref: '#/components/examples/post-messages-by-address-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /outputs/by-hash:
+    get:
+      tags:
+        - transaction-messages
+      summary: Find an output by its hash.
+      description: Find an output by its hash. Get the producer of the output, the corresponding address, amount and spend status. This information can only be retrieved for outputs which are part of a confirmed transaction.
+      operationId: get-outputs-by-hash
+      parameters:
+        - in: query
+          name: hashes
+          schema:
+            type: array
+            items:
+              type: string
+          required: true
+          description: Hash of the output.
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetOutputsByHashResponse"
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/post-outputs-by-hash-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /outputs/by-address:
+    get:
+      tags:
+        - transaction-messages
+      summary: Get all outputs that use a given address.
+      description: Get all outputs that use a given address.
+      operationId: get-outputs-by-address
+      parameters:
+        - in: query
+          name: outputs
```

Comment:
The query parameter should be `addresses` since one is passing addresses and not output ids.